superset/dashboards/api.py

```diff
@@ -341,7 +346,13 @@ def get(
             404:
               $ref: '#/components/responses/404'
         """
-        result = self.dashboard_get_response_schema.dump(dash)
+        schema = (
+            self.dashboard_get_response_schema_guest
+            if security_manager.is_guest_user()
+            else self.dashboard_get_response_schema
```

Comment:
@jfrag1 I think we can even move the schema logic into Marshmallow dynamically with either a type function or a   @marshmallow.post_dump decorator. That way the logic isn't in the api layer and you won't need two schemas.